src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -81,6 +81,12 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
                     $otherViews
                 );
             }
+
+            if ($groupByOrder && is_array($groupByOrder)) {
+                uksort($otherViews, function ($key1, $key2) use ($groupByOrder) {
```

Comment:
The callable should be compatible with `uksort` - anything that takes two parameters, compares them, and returns an integer.  Your code would then look something like this:

```
if ($groupByOrder) {
    if (is_callable($groupByOrder)) {
        uksort($otherViews, $groupByOrder);
    } elseif (is_array($groupByOrder)) {
        // TODO: Your existing code goes here
    }
}
```

I could then use any type of callable for the `groupByOrder` option:

 - `'strnatcmp'` - Sort them using a "natural order" algorithm via the `strnatcmp()` function
 - `[$this, 'sortGroups']` - Use the `sortGroups()` function on `$this`
 - `[Foo::class, 'sortGroups']` - Use the static `sortGroups()` function from the `Foo` class
 - `function ($a, $b) { return $b <=> $a; }` - Use some anonymous function